tests/test_core.py

```diff
@@ -0,0 +1,237 @@
+# type: ignore
+from invoke import MockContext
+
+import yaml
+from pytest import fixture 
+from textwrap import dedent
+
+import gitops.utils.kube as kube
+import gitops.utils.apps as apps
+from gitops import core
+
+
+@fixture(autouse=True)
+def mock_invoke_run(monkeypatch):
+    monkeypatch.setattr(core, "run", lambda cmd: None)
+
+@fixture
+def confirm_yes(monkeypatch):
+    monkeypatch.setattr(apps, "confirm", lambda: True)
+
+@fixture
+def sample_app(tmp_path, monkeypatch):
+    monkeypatch.setattr(apps, "get_apps_directory", lambda: tmp_path)
+    (tmp_path / "sample_app").mkdir(parents=True, exist_ok=True)
+    (tmp_path / "sample_app/deployment.yml").write_text("{}")
+    (tmp_path / "sample_app/secrets.yml").write_text("{}")
+    return tmp_path / "sample_app/"
+
+
+class TestMixin:
+    @fixture
+    def deployment_yml(self, sample_app):
+        def _deployment_yml(content):
+            content = dedent(content)
+            path = sample_app / "deployment.yml"
+            path.write_text(content)
+            self.original_content = content
+            self.deployment_yml_path = path
+            return path
+
+        return _deployment_yml
+
+    def deployment_yml_is_unchanged(self):
+        return self.original_content == self.deployment_yml_path.read_text()
+
+
+class TestSetEnv(TestMixin):
+    def test_setenv_should_not_overwrite_if_there_is_existing_variable(
+        self,
+        deployment_yml,
+        confirm_yes,
+    ):
+        file = deployment_yml(
+            """\
+            namespace: test
+            chart: test
+            environment:
+              ANOTHER: WORLD
+              HELLO: PARTNER
+            """
+        )
+        core.setenv(MockContext(), filter="all", values="HELLO=WORLD")
+        assert self.deployment_yml_is_unchanged()
```

Comment:
It seems like `setenv` does not modify any existing deployments if they already have an env variable with that key? That seems like an unexpected behavior, but that's the pre-existing behavior.